tests/data/CMakeLists.txt

```diff
@@ -98,3 +99,8 @@ add_custom_command(
   DEPENDS ${BTF_DATA_HEX} ${FUNC_LIST_HEX} ${CONFIGURE_BTF_HEADERS})
 
 add_custom_target(debuginfo_btf_data DEPENDS ${BTF_DATA_H})
+
+# BTF doesn't support C++, so we only generate a data_source_cxx executable
```

Comment:
I understand that this is much easier to do than what we do for `data_source` but I'm not a big fan of using different approaches for generating C and C++ DWARF data. Maybe we could try to unify them somehow?

I wouldn't be against using the same "just use the compiled binary" approach for C dwarf data, too. It's unnecessarily complicated (as it was inspired by the BTF approach which has some specifics) and requires additional tools (`strip`, `xxd`) during the build.